client/consensus/babe/src/lib.rs

```diff
@@ -399,6 +399,9 @@ pub struct BabeParams<B: BlockT, C, E, I, SO, SC, CAW, BS, IDP> {
 	/// Checks if the current native implementation can author with a runtime at a given block.
 	pub can_author_with: CAW,
 
+	/// Hook into the sync module to control the justification sync process.
+	pub justification_sync_link: L,
```

Comment:
We should really rewrite all this stuff to have *one* block import and not this stuff here that is split :see_no_evil:

Reply:
Do you mean having the exact same process for importing locally authored and synced blocks? If that's you mean then I agree as I also don't like having to duplicate this logic from the import queue, but that requires a bigger refactoring 🥲.